go/libraries/doltcore/sqle/dprocedures/dolt_checkout.go

```diff
@@ -476,9 +477,17 @@ func doGlobalCheckout(ctx *sql.Context, branchName string, isForce bool, isNewBr
 }
 
 func checkoutTables(ctx *sql.Context, roots doltdb.Roots, name string, tables []string) error {
-	// TODO: schema name
-	roots, err := actions.MoveTablesFromHeadToWorking(ctx, roots, doltdb.ToTableNames(tables, doltdb.DefaultSchemaName))
+	var err error
+	tableNames := make([]doltdb.TableName, len(tables))
 
+	for i, table := range tables {
+		tableNames[i], _, _, err = resolve.Table(ctx, roots.Working, table)
```

Comment:
This will put an empty TableName in the list if the table doesn't exist in working, which is not what you want. Annoyingly, there's a bunch of logic in `MoveTablesFromHeadToWorking` that checks STAGED, then HEAD for a table with this name. So what you actually want is a function that resolves a table by looking in all three roots (working, staged, head) in that order. Should probably live in the actions package next to `MoveTablesFromHeadToWorking`, call it `FindTableInRoots(string, doltdb.Roots)`. Same return values as `resolve.Table`.